### PR TITLE
spatial/r1: import and rename from bound

### PR DESCRIPTION
**Pull request by sbinet**
Mechanically generated with:

```
$> gomvpkg -from gonum.org/v1/gonum/bound -to gonum.org/v1/gonum/spatial/r1 -vcs_mv_cmd "git mv {{.Src}} {{.Dst}}"
```

Fixes gonum/gonum#956.

Please take a look.

### CODE REVIEW

**Comment by sbinet** [comment]
ah... perhaps hold this one off before https://github.com/gonum/gonum/pull/969 has been sorted out and/or merged in?

**Comment by kortschak** [comment]
We should probably wait. Also, what do you think of `r1.Interval`? The reasoning for the question is that a bound is a semantic layer on top of an interval.

**Comment by sbinet** [comment]
I am fine with `s/Bound/Interval/` (probably with a `gorename`.)
I'll wait for `#969` to be collected.

**Comment by vladimir-ch** [comment]
> ... bound is a semantic layer on top of an interval

I also thought about this and this may also be where the confusion about `bound.Union` comes from. I like how `r2.Box` and `r3.Box` provide only the type for storing two points without implying any semantics (is the boundary included, what is an empty box, does it bound anything, ...).

**Comment by kortschak** [comment]
Yes, I'm beginning to regret https://github.com/gonum/gonum/pull/845#discussion_r255359176. I think perhaps those additions should be rolled back.

**Comment by kortschak** [comment]
@sbinet The PRs arising from #969 have been reverted, so this can go ahead now. Do you want to do that?

**Comment by sbinet** [comment]
Sure. I'll update this PR shortly

**Comment by sbinet** [comment]
PTAL

File: spatial/r1/doc.go [hunk 1]
```diff
@@ -2,5 +2,5 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-// Package bound provides types for bounded data.
-package bound // import "gonum.org/v1/gonum/bound"
+// Package r1 provides 1D vectors and boxes and operations on them.
```
Comment (kortschak) [inline spatial/r1/doc.go]: s/boxes/intervals/.

File: spatial/r1/bound.go [hunk 2]
```diff
@@ -2,9 +2,9 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-package bound
+package r1
 
-// Bound represents [Min, Max] bounds.
-type Bound struct {
+// Interval represents [Min, Max] bounds.
```
Comment (kortschak) [inline spatial/r1/bound.go]: @vladimir-ch Do we want to be this definitive? We use them as [min, max] bounds, but they could be any type of real interval.

Comment (vladimir-ch) [reply spatial/r1/bound.go]: I would prefer leaving it unspecified, it was the main motivation behind these changes. But what will we put there instead? `Interval is an interval`? :-)

Comment (kortschak) [reply spatial/r1/bound.go]: Simple, direct and to the point. SGTM ("Interval is an interval in R1" seems like over cooking.)

**Comment by sbinet** [comment]
PTAL

**Pull request merged**